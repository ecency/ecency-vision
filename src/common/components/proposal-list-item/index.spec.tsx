import React from "react";

import renderer from "react-test-renderer";

import { createBrowserHistory, createLocation } from "history";
import { StaticRouter } from "react-router-dom";

import {
  globalInstance,
  UiInstance,
  proposalInstance,
  dynamicPropsIntance1,
  activeUserMaker,
  allOver
} from "../../helper/test-helper";

import { ProposalListItem } from "./index";

jest.mock("../../util/now", () => () => new Date("November 22, 2020 03:24:00"));

jest.mock("../../api/hive", () => ({
  getProposalVotes: (proposalId: number, voter: string = "", limit: number = 300) =>
    new Promise((resolve) => {
      resolve([]);
    })
}));

const defProps = {
  history: createBrowserHistory(),
  location: createLocation({}),
  global: globalInstance,
  dynamicProps: dynamicPropsIntance1,
  users: [],
  activeUser: activeUserMaker("foo"),
  ui: UiInstance,
  signingKey: "",
  addAccount: () => {},
  proposal: proposalInstance,
  setActiveUser: () => {},
  updateActiveUser: () => {},
  deleteUser: () => {},
  toggleUIProp: () => {},
  setSigningKey: () => {}
};

it("(1) Default render.", async () => {
  const props = { ...defProps };
  const component = renderer.create(
    <StaticRouter location="/" context={{}}>
      <ProposalListItem {...props} />
    </StaticRouter>
  );
  await allOver();
  expect(component.toJSON()).toMatchSnapshot();
});

// Return proposal is dynamic now, disabled this test.
/*it('(2) Return proposal.', async () => {
    const props = {
        ...defProps,
        isReturnProposalId: 0,
        proposal: {...proposalInstance, id: 0, proposal_id: 0}
    }

    const component = await renderer.create(
        <StaticRouter location="/" context={{}}>
            <ProposalListItem {...props}/>
        </StaticRouter>);
    await allOver();
    expect(component.toJSON()).toMatchSnapshot();
});*/
