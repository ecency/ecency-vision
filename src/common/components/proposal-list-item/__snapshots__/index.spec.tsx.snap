// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render. 1`] = `
<div
  className="proposal-list-item"
>
  <div
    className="item-content"
  >
    <div
      className="left-side"
    >
      <div
        className="proposal-users-card"
      >
        <span
          className="user-avatar small"
          onClick={[Function]}
          style={
            Object {
              "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/small)",
            }
          }
        />
        <span
          className="users"
        >
          by
           
          <a
            href="/@good-karma"
            onClick={[Function]}
          >
            <span>
               
              good-karma
            </span>
          </a>
           
          for
           
          <a
            href="/@ecency"
            onClick={[Function]}
          >
            <span>
               
              ecency
            </span>
          </a>
        </span>
      </div>
      <div
        className="proposal-title"
      >
        <a
          href="/proposals/88"
          onClick={[Function]}
        >
          Hivesigner - Ongoing development and improvement
           
          <span
            className="proposal-id"
          >
            #
            88
          </span>
        </a>
      </div>
      <div
        className="status-duration-payment"
      >
        <div
          className="proposal-status active"
        >
          active
        </div>
        <div
          className="proposal-duration"
        >
          Apr 15, 2020
           
          -
           
          Apr 15, 2022
           (
          730 days
          )
        </div>
        <div
          className="proposal-payment"
        >
          <span
            className="all-pay"
          >
            51.1k HBD
          </span>
          <span
            className="daily-pay"
          >
            (
            Daily 70.0
             
            HBD
            )
          </span>
        </div>
      </div>
      <div
        className="permlink"
      >
        <a
          href="/proposal/@good-karma/hivesigner-ongoing-development-and-improvement"
          onClick={[Function]}
        >
          <svg
            viewBox="0 0 24 24"
          >
            <path
              d="M3.9,12C3.9,10.29 5.29,8.9 7,8.9H11V7H7A5,5 0 0,0 2,12A5,5 0 0,0 7,17H11V15.1H7C5.29,15.1 3.9,13.71 3.9,12M8,13H16V11H8V13M17,7H13V8.9H17C18.71,8.9 20.1,10.29 20.1,12C20.1,13.71 18.71,15.1 17,15.1H13V17H17A5,5 0 0,0 22,12A5,5 0 0,0 17,7Z"
              fill="currentColor"
            />
          </svg>
           
          /
          good-karma
          /
          hivesigner-ongoing-development-and-improvement
        </a>
      </div>
      <div
        className="votes"
      >
        <a
          className="btn-votes"
          href="#"
          onClick={[Function]}
        >
          Votes: 30,331,489.512 HP
        </a>
      </div>
    </div>
    <div
      className="right-side"
    >
      <div
        className="voting"
      >
        <div
          className="proposal-vote-btn"
          onClick={[Function]}
        >
          <div
            className="btn-proposal-vote btn-up-vote vote-btn-lg in-progress"
          >
            <span
              className="btn-inner"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        className="remaining-days"
      >
        Remaining: 508 days
      </div>
    </div>
  </div>
</div>
`;
